vm/src/obj/objdict.rs

```diff
@@ -205,14 +205,31 @@ impl PyDictRef {
 
     #[cfg_attr(feature = "flame-it", flame("PyDictRef"))]
     fn inner_getitem(self, key: PyObjectRef, vm: &VirtualMachine) -> PyResult {
-        if let Some(value) = self.entries.borrow().get(vm, &key)? {
-            return Ok(value);
+        if let Some(value) = self.inner_getitem_option(&key, vm)? {
+            Ok(value)
+        } else {
+            Err(vm.new_key_error(key.clone()))
+        }
+    }
+
+    /// Return an optional inner item, or an error (can be key error as well)
+    fn inner_getitem_option<K: DictKey + IntoPyObject + Clone>(
+        &self,
+        key: &K,
```

Comment:
I'd say to accept just a `K` in order to avoid a double allocation when you `key.clone()`.

Reply:
Yes, thank you, I did this, and it worked nicely! Now we can also pass `&str` values to these functions!